### PR TITLE
Integration Test Error Collection

### PR DESCRIPTION
**Pull request by nholland94**
This PR contains work for both #5576 and #5591.

Tested using https://github.com/CodaProtocol/coda/pull/5657 as a deployment target. Tests successfully collect error and warning logs, and are able to gracefully recover from signals, fatal logs, exceptions, and or_error return values.

### CODE REVIEW

File: src/lib/cache_dir/cache_dir.ml [hunk 1]
```diff
@@ -41,7 +41,10 @@ let possible_paths base =
   List.map [env_path; brew_install_path; s3_install_path; autogen_path]
     ~f:(fun d -> d ^/ base)
 
-let load_from_s3 s3_bucket_prefix s3_install_path ~logger =
+let load_from_s3 _s3_bucket_prefix _s3_install_path ~logger:_ =
+  Deferred.return (Error (Error.of_string "temporarily disabled"))
```
Comment (psteckler) [inline src/lib/cache_dir/cache_dir.ml:45]: is this change meant for the PR?

Comment (nholland94) [reply src/lib/cache_dir/cache_dir.ml:45]: Yes, my bad. Removing.

File: src/lib/integration_test_lib/integration_test_lib.ml [hunk 2]
```diff
@@ -33,6 +33,73 @@ module Test_config = struct
...
+        occurrence_time
+
+  module Set = struct
+    type nonrec t = {soft_errors: t list; hard_errors: t list}
+
+    let empty = {soft_errors= []; hard_errors= []}
+
+    let soft_singleton err = {empty with soft_errors= [err]}
+
+    let hard_singleton err = {empty with soft_errors= [err]}
```
Comment (psteckler) [inline src/lib/integration_test_lib/integration_test_lib.ml:68]: did you mean `hard_errors=...` here?

File: src/lib/key_cache/key_cache.ml [hunk 3]
```diff
@@ -89,7 +89,12 @@ module Sync : S with module M := Or_error = struct
     in
     {read; write}
 
-  let s3 to_string read ~bucket_prefix ~install_path =
+  let s3 _to_string _read ~bucket_prefix:_ ~install_path:_ =
```
Comment (psteckler) [inline src/lib/key_cache/key_cache.ml]: is this meant for the PR?

File: src/app/test_executive/test_executive.ml [hunk 4]
```diff
@@ -11,10 +11,38 @@ let tests : test list =
...
+        let color =
+          match error_type with
+          | `Soft ->
+              Bash_colors.yellow
+          | `Hard ->
+              Bash_colors.red
+        in
+        Print.eprintf "    %s%s%s\n" color (to_string error) Bash_colors.none
+    ) ;
+    Out_channel.(flush stdout) ;
```
Comment (psteckler) [inline src/app/test_executive/test_executive.ml:41]: should the flush be on `stderr`?

Comment (nholland94) [reply src/app/test_executive/test_executive.ml:41]: Yes, thank you for catching that.

File: src/lib/integration_test_lib/integration_test_lib.ml (hunk 2 above)
Comment (nholland94) [reply src/lib/integration_test_lib/integration_test_lib.ml:68]: oops

File: src/lib/integration_test_cloud_engine/stack_driver_log_engine.ml [hunk 5]
```diff
@@ -375,113 +422,193 @@ module Block_produced_query = struct
...
+    t
+  in
+  let%bind () = Deferred.map (cancel_background_tasks ()) ~f:Or_error.return in
+  let%map () =
+    let open Deferred.Let_syntax in
+    match%map delete_subscriptions subscriptions with
+    | Ok _ ->
+        Ok ()
+    | Error e' ->
+        [%log' fatal logger] "Error deleting subscriptions: $error"
```
Comment (psteckler) [inline src/lib/integration_test_cloud_engine/stack_driver_log_engine.ml:596]: you can use `%log` and omit `logger` here

File: src/lib/integration_test_cloud_engine/stack_driver_log_engine.ml [hunk 6]
```diff
@@ -375,113 +422,193 @@ module Block_produced_query = struct
...
-  [%log' info logger] "Pulling initialization subscription" ;
   let%bind results =
     uninterruptible
       (let open Deferred.Or_error.Let_syntax in
-      let%bind logs = Subscription.pull initialization_subscription in
-      Deferred.return (or_error_list_map logs ~f:Initialization_query.parse))
+      let%bind logs = Subscription.pull subscription in
+      Deferred.return (or_error_list_map logs ~f:parse_subscription))
   in
+  [%log' info logger] "Pulling %s subscription" subscription_name ;
```
Comment (psteckler) [inline src/lib/integration_test_cloud_engine/stack_driver_log_engine.ml:471]: `[%log info]`

File: src/app/test_executive/test_executive.ml [hunk 7]
```diff
@@ -35,19 +63,98 @@ let main inputs =
...
+        | Some log_engine -> (
+            match%map Engine.Log_engine.destroy log_engine with
+            | Ok errors ->
+                errors
+            | Error err ->
+                { Test_error.Set.empty with
+                  hard_errors= [Test_error.internal_error err] } )
+      in
+      let%bind () =
+        match !net_manager_ref with
```
Comment (psteckler) [inline src/app/test_executive/test_executive.ml:93]: could use `Option.map` here ... or not

Comment (psteckler) [inline src/app/test_executive/test_executive.ml:93]: could use `Option.value_map` here

**Pull request merged**